cogs/games.py

```diff
@@ -23,15 +23,182 @@
+            return
+        category: str = command or random.choice(list(
+            self.hangman_dict.keys()))
+        try:
+            word_list, pretty_name = self.hangman_dict[category]
+        except KeyError:
+            cat_list: str = ", ".join(
+                f"`{hm_cat}` (length: {len(self.hangman_dict[hm_cat][0])})"
+                for hm_cat in sorted(self.hangman_dict.keys()))
+            await ctx.send(f"command `{command}` is not valid\n"
+                           f"here is a list of valid commands: {cat_list}")
+            return
+
+        hm_word, hm_img = random.choice(word_list)
+        lowered_word = hm_word.lower()
+
+        not_guessed: Set[str] = set(re.sub(r"[^a-z]", "", lowered_word))
+        incorrect_guesses: Set[str] = set()
+        timeout_dict: Dict[discord.Member, float] = {}
+        winner: discord.Member = None
+        cool_win: bool = False
+
+        def wait_for_check(msg) -> bool:
+            return msg.channel == ctx.message.channel and (
+                (len(msg.content) == 1 and msg.content.isalpha())
+                or msg.content.lower() == lowered_word)
+
+        update_embed, hm_embed = mk_hm_embed_up_fn(pretty_name, hm_word,
+                                                   lowered_word, not_guessed,
+                                                   incorrect_guesses)
+        await ctx.send(embed=hm_embed)
+
+        while True:
+
+            try:
+                curr_msg = await self.bot.wait_for('message',
+                                                   check=wait_for_check,
+                                                   timeout=self.hm_timeout)
+            except TimeoutError:
+                update_embed("the game has timed out")
+                await ctx.send(embed=hm_embed)
+                await ctx.send(
+                    f"sorry everyone, no one has interacted with the "
+                    f"hangman in {self.hm_timeout//60} minutes, "
+                    f"the game has timed out")
+                return
+
+            curr_guess = curr_msg.content.lower()
+            if not (curr_msg.author in timeout_dict and
+                    (time() - timeout_dict[curr_msg.author]) < 3.0
+                    ):    # check that user isn't time'd out
+                if curr_guess == lowered_word:
+                    not_guessed = set()
+                    winner = curr_msg.author
+                    cool_win = len(not_guessed) > (len(set(lowered_word)) //
+                                                   2.5)
+                    update_embed(f"{winner} guessed the entire word!",
+                                 correct_guess=True,
+                                 img_url=hm_img)
+                    await ctx.send(embed=hm_embed)
+                    await ctx.send(
+                        f"congratulations {winner}, you solved the hangman"
+                        f"{', but in a cool way' if  cool_win else ''}, "
```

Comment:
"but in a cool way" i'd say isn't ideal grammar here - maybe "[...] you solved the hangman (in a cool way 😎), earning you [...]" or something like that